src/cpp/shared/scripting/scripting.h

```diff
@@ -5,6 +5,12 @@
 #include<map>
 #include<string>
 
+extern "C" {
+#include "lua/src/lua.h"
+#include "lua/src/lualib.h"
```

Comment:
Only the first include is needed.

And really, it's not needed. You can just write "struct lua_State;", and it will do all you need. No need to make someone have to think about why you're importing these things.